tests/neteventebpfext/neteventebpfext_unit/netevent_ebpfext_unit.cpp

```diff
@@ -260,3 +261,81 @@ TEST_CASE("netevent_drivers_load_unload_stress", "[neteventebpfext]")
     REQUIRE(neteventebpfext_driver.stop() == true);
     REQUIRE(neteventebpfext_driver.unload() == true);
 }
+
+TEST_CASE("netevent_bpf_prog_run_test", "[neteventebpfext]")
+{
+    // Free the BPF object will take some time to unload from the previous test
+    // Once this issue is fixed, the sleep can be removed: https://github.com/microsoft/ebpf-for-windows/issues/2667
+    std::this_thread::sleep_for(std::chrono::seconds(10));
+
+    // Load and start neteventebpfext extension driver.
+    driver_service neteventebpfext_driver;
+    REQUIRE(
+        neteventebpfext_driver.create(
+            L"neteventebpfext", driver_service::get_driver_path("neteventebpfext.sys").c_str()) == true);
+    REQUIRE(neteventebpfext_driver.start() == true);
+
+    // Load the NetEventMonitor native BPF program.
+    struct bpf_object* object = bpf_object__open("netevent_monitor.sys");
+    REQUIRE(object != nullptr);
+
+    int res = bpf_object__load(object);
+    REQUIRE(res == 0);
+
+    // Find and attach to the netevent_monitor BPF program.
+    auto netevent_monitor = bpf_object__find_program_by_name(object, "NetEventMonitor");
+    REQUIRE(netevent_monitor != nullptr);
+    auto netevent_monitor_link = bpf_program__attach(netevent_monitor);
+    REQUIRE(netevent_monitor_link != nullptr);
+
+    // Attach to the eBPF ring buffer event map.
+    bpf_map* netevent_events_map = bpf_object__find_map_by_name(object, "netevent_events_map");
+    REQUIRE(netevent_events_map != nullptr);
+    auto ring = ring_buffer__new(bpf_map__fd(netevent_events_map), netevent_monitor_event_callback, nullptr, nullptr);
+    REQUIRE(ring != nullptr);
+
+
+    // Initialize structures required for bpf_prog_test_run_opts
+    bpf_test_run_opts bpf_opts = {0};
+    netevent_event_md_t netevent_md = {0};
+    unsigned char dummy_data_in[] = {NOTIFY_EVENT_TYPE_NETEVENT, 'a', 'b'};
+    const size_t dummy_data_size = sizeof(dummy_data_in);
+    unsigned char data_out[MAX_PACKET_SIZE] = {0};
+    uint32_t event_count_before = event_count;
+    fd_t netevent_program_fd = bpf_program__fd(netevent_monitor);
+
+    // Prepare bpf_opts
+    bpf_opts.repeat = 1;
+    bpf_opts.ctx_in = &netevent_md;
+    bpf_opts.ctx_size_in = sizeof(netevent_md);
+    bpf_opts.ctx_out = nullptr;
+    bpf_opts.ctx_size_out = 0;
+    bpf_opts.data_in = &dummy_data_in;
+    bpf_opts.data_size_in = static_cast<uint32_t>(dummy_data_size);
+    // Set the data_out buffer to hold the output.
+    bpf_opts.data_out = data_out;
+    bpf_opts.data_size_out = sizeof(data_out);
+
+    // Execute the program
+    REQUIRE(bpf_prog_test_run_opts(netevent_program_fd, &bpf_opts) == 0);
```

Comment:
Suggest: Consider adding some negative test cases (i.e where the bpf_opts params are invalid, and ensure that the bpf_prog_test_run_opts returns appropriate errors)